### PR TITLE
Move modal outside of the Version component.

### PR DESCRIPTION
**Pull request by artonge**
This is for accessibility, to have the `NcListItem` (`<li>`) as a direct child of the `<ul>`

Before | After
-- | --
image | Screenshot from 2024-01-24 15-48-13

Diff is big, but is mostly white spaces, moved code, and a rewrite in ts of `versions.js`.

### CODE REVIEW

**Review by JuliaKirschenheuter — APPROVED** [review]
Thank you Louis!

**Review by emoral435 — APPROVED** [review]

**Review by nfebe — APPROVED** [review]
Looks good!

**Review by ShGKme — CHANGES_REQUESTED** [review]

File: apps/files_versions/src/views/VersionTab.vue [hunk 1]
```diff
@@ -16,30 +16,37 @@
...
-						@restore="handleRestore"
-						@label-update="handleLabelUpdate"
-						@delete="handleDelete" />
-				</template>
-			</ul>
-		</template>
-		<NcLoadingIcon v-if="loading" slot="loader" class="files-list-viewer__loader" />
-	</VirtualScrolling>
+	<div>
+		<VirtualScrolling :sections="sections"
```
[the remark this answers is not in the document]
Comment (ShGKme) [reply apps/files_versions/src/views/VersionTab.vue:20]: Before | After
---|---
image | image
image | image

[the remark this answers is not in the document]
Comment (artonge) [reply apps/files_versions/src/views/VersionTab.vue]: Good catch, thanks.
Fixed

**Review by ShGKme — APPROVED** [review]

**Comment by susnux** [comment]
cypress is related and needs fixing

**Pull request merged**